pkg/api/types.go

```diff
@@ -1042,6 +1042,26 @@ type AzureFileVolumeSource struct {
 	// the ReadOnly setting in VolumeMounts.
 	// +optional
 	ReadOnly bool
+	// the namespace of the secret that contains Azure Storage Account Name and Key
+	// default is the same as the Pod
+	// +optional
+	SecretNamespace *string
```

Comment:
don't add this field to AzureFileVolumeSource, you should only be able to specify a namespace when dealing with a PV volume source